src/randoop/JunitFileWriter.java

```diff
@@ -195,58 +183,86 @@ public File writeDriverFile(String driverClassName) {
+      out.println("import org.junit.runner.RunWith;");
+      out.println("import org.junit.runners.Suite;");
+      out.println();
+      out.println("@RunWith(Suite.class)");
+      out.println("@Suite.SuiteClasses({");
+     
+      Iterator<String> testIterator = testClassNames.iterator();
+      if (testIterator.hasNext()) {
+        out.println(testIterator.next() + ".class");
+        while (testIterator.hasNext()) {
+          out.print(", ");
+          out.println(testIterator.next() + ".class");
+        }
       }
-      if (GenInputsAbstract.init_routine != null)
-        out.println ("    " + GenInputsAbstract.init_routine + "();");
-
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("    TestRunner runner = new TestRunner();");
-      out.println("    TestResult result = runner.doRun(suite(), false);");
-      out.println("    if (! result.wasSuccessful()) {");
-      out.println("      System.exit(1);");
-      out.println("    }");
-      } else {
-      out.println("    new " + driverClassName + "(\"" + driverClassName + "\").runTest();");
+      
+      out.println("})");
+      out.println("public class " + suiteClassName + "{ }");
+    } finally {
+      if (out != null) {
+        out.close();
       }
-      out.println("  }");
-      out.println("");
-      out.println("  public " + driverClassName + "(String name) {");
-      out.println("    super(name);");
-      out.println("  }");
-      out.println("");
-      if (GenInputsAbstract.junit_reflection_allowed) {
-      out.println("  public static Test suite() {");
-      out.println("    TestSuite result = new TestSuite();");
-      for (String junitTestsClassName : junitTestSuiteNames) {
-        out.println("    result.addTest(new TestSuite(" + junitTestsClassName + ".class));");
+    }
+    return file;
+  }
+  
+  /*
+   * write non-reflective driver for tests
+   * Note that this relies on method names being generated as "test"+i in writeSubSuite
+   */
+  public File writeDriverFile(List<String> testClassNames) {
+    File dir = this.getDir();
+    File file = new File(dir, masterTestClassName + "Driver.java");
+    PrintStream out = createTextOutputStream(file);
+    try {
+      outputPackageName(out, packageName);
+
+      out.println();
+      out.println("public class " + masterTestClassName + "Driver {");
+      out.println();
+      out.print("  public static void main(String[] args) {");
+      
+      if (GenInputsAbstract.init_routine != null) {
+        out.println ("    " + GenInputsAbstract.init_routine + "();");
       }
-      out.println("    return result;");
-      out.println("  }");
-      } else {
-      out.println("  @Override");
-      out.println("  public void runTest() throws Throwable {");
-      for (String junitTestsClassName : junitTestSuiteNames) {
-        out.println("    new " + junitTestsClassName + "().runTest();");
+      
+      int classIndex = 0; //counter for test class object instances
```

Comment:
I greatly prefer for-each loops to integral-bounded for loops.
However, I think it's even more important to keep all the loop iteration logic together and simple, so I would probably use a foreach loop here.  Otherwise, the reader will have to skim the code to understand how classindex is being updated.  I like to reserve loop-carried dependencies for unusual situations; they signal that something unusual is going on.